private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -61,39 +206,77 @@ export class TemplateServiceClient {
         const newTemplate: ITemplate = {
             instances: [templateInstance],
             tags: [],
-            owner: owner,
-            isPublished: false
+            owner: this.ownerID!,
+            isPublished: isPublished
         }
 
-        return TemplateServiceClient.storageProvider.insertTemplate(newTemplate);
+        return this.storageProvider.insertTemplate(newTemplate);
     }
 
     /**
      * @public
      * Get entry point. 
+     * Returns specified templates
      * @param {string} templateId - unique template id
      * @param {boolean} isPublished 
      * @param {string} templateName - name to query for
      * @param {string} version - version number
+     * @param {boolean} owned - If false, will retrieve all public templates regardless of owner
      * @returns Promise as valid json 
      */
-    public async getTemplates(templateId?: string, isPublished?: boolean, templateName?: string, version?: number): Promise<JSONResponse<ITemplate[]>> {
-        let owner = TemplateServiceClient.authProvider.getOwner();
+    public async getTemplates(templateId?: string, isPublished?: boolean, templateName?: string, version?: number, owned?: boolean): Promise<JSONResponse<ITemplate[]>> {
```

Comment:
is templateId uniquely identifies template? If that is the  case the method name should be getTemplate?